internal/gengapic/testdata/method_GetAnotherThing.want

```diff
@@ -6,39 +6,39 @@ func (c *fooGRPCClient) GetAnotherThing(ctx context.Context, req *mypackagepb.In
-	headerName1 := "name"
-	headerValue1 := ""
-	if routingRegexp1.MatchString(req.GetOther()){headerValue1 = url.QueryEscape(routingRegexp1.FindStringSubmatch(req.GetOther())[1])}
-	if len(headerValue1) > 0{routingHeadersMap[headerName1] = headerValue1}
-	routingRegexp2 := regexp.MustCompile("(?P<foo_name>projects/[^/]+)/bars/[^/]+(?:/.*)?")
-	headerName2 := "foo_name"
-	headerValue2 := ""
-	if routingRegexp2.MatchString(req.GetAnother()){headerValue2 = url.QueryEscape(routingRegexp2.FindStringSubmatch(req.GetAnother())[1])}
-	if len(headerValue2) > 0{routingHeadersMap[headerName2] = headerValue2}
-	routingRegexp3 := regexp.MustCompile("(?P<foo_name>projects/[^/]+/foos/[^/]+)/bars/[^/]+(?:/.*)?")
-	headerName3 := "foo_name"
-	headerValue3 := ""
-	if routingRegexp3.MatchString(req.GetAnother()){headerValue3 = url.QueryEscape(routingRegexp3.FindStringSubmatch(req.GetAnother())[1])}
-	if len(headerValue3) > 0{routingHeadersMap[headerName3] = headerValue3}
-	routingRegexp4 := regexp.MustCompile("(?P<nested_name>.*)")
-	headerName4 := "nested_name"
-	headerValue4 := ""
-	if routingRegexp4.MatchString(req.GetFieldName().GetNested()){headerValue4 = url.QueryEscape(routingRegexp4.FindStringSubmatch(req.GetFieldName().GetNested())[1])}
-	if len(headerValue4) > 0{routingHeadersMap[headerName4] = headerValue4}
-	routingRegexp5 := regexp.MustCompile("(?P<part_of_nested>projects/[^/]+)/bars")
-	headerName5 := "part_of_nested"
-	headerValue5 := ""
-	if routingRegexp5.MatchString(req.GetFieldName().GetNested()){headerValue5 = url.QueryEscape(routingRegexp5.FindStringSubmatch(req.GetFieldName().GetNested())[1])}
-	if len(headerValue5) > 0{routingHeadersMap[headerName5] = headerValue5}
-	for headerName, headerValue := range routingHeadersMap {routingHeaders = routingHeaders + fmt.Sprintf("%s=%s&", headerName, headerValue)}
-	if len(routingHeaders) > 0{routingHeaders = routingHeaders[:len(routingHeaders)-1]}
+	if reg := regexp.MustCompile("(.*)");
+	reg.MatchString(req.GetOther()) {
+		routingHeadersMap["other"] = url.QueryEscape(reg.FindStringSubmatch(req.GetOther())[1])
+	}
+	if reg := regexp.MustCompile("(?P<name>projects/[^/]+)/foos");
+	reg.MatchString(req.GetOther()) {
+		routingHeadersMap["name"] = url.QueryEscape(reg.FindStringSubmatch(req.GetOther())[1])
+	}
+	if reg := regexp.MustCompile("(?P<foo_name>projects/[^/]+)/bars/[^/]+(?:/.*)?");
+	reg.MatchString(req.GetAnother()) {
+		routingHeadersMap["foo_name"] = url.QueryEscape(reg.FindStringSubmatch(req.GetAnother())[1])
+	} else if reg = regexp.MustCompile("(?P<foo_name>projects/[^/]+/foos/[^/]+)/bars/[^/]+(?:/.*)?");
+	reg.MatchString(req.GetAnother()) {
+		routingHeadersMap["foo_name"] = url.QueryEscape(reg.FindStringSubmatch(req.GetAnother())[1])
+	}
+	if reg := regexp.MustCompile("(?P<foo_name>projects/[^/]+/foos/[^/]+)/bars/[^/]+(?:/.*)?");
+	reg.MatchString(req.GetAnother()) {
+		routingHeadersMap["foo_name"] = url.QueryEscape(reg.FindStringSubmatch(req.GetAnother())[1])
+	}
```

Comment:
I think this should be an `else if` connected to the `if`/`else if` immediately above, yeah?

Reply:
I think just straight `if` statements might be better actually, so let's stick with what you have.